argcomplete/__init__.py

```diff
@@ -447,6 +532,86 @@ def complete(self, text, state):
         else:
             return None
 
+    def rl_complete(self, text, state):
+        """
+        A complete function for readline.
+        Usage:
+
+        .. code-block:: python
+
+            import argcomplete, argparse, readline
+            parser = argparse.ArgumentParser()
+            ...
+            completer = argcomplete.CompletionFinder(parser)
+            readline.set_completer(completer.rl_complete)
+            readline.set_completer_delims(' \t\n=')
+            readline.parse_and_bind('"\M-OA": previous-history')
+            readline.parse_and_bind("tab: complete")
+            result = input("prompt> ")
+
+        (Use ``raw_input`` instead of ``input`` on Python 2, or use `eight <https://github.com/kislyuk/eight>`_).
+        """
+        debug('text:[{0}]'.format(text), ', state:', state)
+        if state == 0:
+            # `text` may empty, so we read line from readline.
+            import readline
+            # ignore word after tab-complete scope.
+            line = readline.get_line_buffer()[:readline.get_endidx()]
+
+            cword_prequote, cword_prefix, cword_suffix, comp_words, first_colon_pos = split_line(line)
+            comp_words.insert(0, sys.argv[0])
+            self.matches = self._get_completions(comp_words, cword_prefix, cword_prequote, first_colon_pos)
+
+            # Complete for `list-node` with input `list-`
+            # but get unexpected result `list-list-node`
+            trim_len = len(cword_prefix) - len(text)
+            if trim_len > 0:
+                self.matches = [x[trim_len:] for x in self.matches]
+
+            debug('matches', self.matches)
+
+        if state < len(self.matches):
+            return self.matches[state]
+        else:
+            return None
+
+    def get_display_completions(self):
+        """
+        This function return complete key and description for customize display hook.
```

Comment:
Please update this docstring with a more specific description of the returned contents. e.g. "This function returns a mapping of option names to their help strings for displaying to the user."